src/Storages/MergeTree/ReplicatedMergeTreeQueue.cpp

```diff
@@ -1751,15 +1749,38 @@ ReplicatedMergeTreeMergePredicate::ReplicatedMergeTreeMergePredicate(
 
     merges_version = queue_.pullLogsToQueue(zookeeper);
 
+    /// Load current quorum status.
+    auto quorum_status_future = zookeeper->asyncTryGet(queue.zookeeper_path + "/quorum/status");
     Coordination::GetResponse quorum_status_response = quorum_status_future.get();
     if (quorum_status_response.error == Coordination::Error::ZOK)
     {
         ReplicatedMergeTreeQuorumEntry quorum_status;
         quorum_status.fromString(quorum_status_response.data);
-        inprogress_quorum_part = quorum_status.part_name;
+        inprogress_quorum_parts.insert(quorum_status.part_name);
     }
     else
-        inprogress_quorum_part.clear();
+        inprogress_quorum_parts.clear();
+
+    Strings partitions = zookeeper->getChildren(queue.replica_path + "/parts");
```

Comment:
This query is quite heavy because we can have a lot of parts. Maybe it's better to list `/blocks` node to get currently inserting quorum parts?